src/app/account/signin/mfa.tsx

```diff
@@ -0,0 +1,75 @@
+import { Panel } from '@/components/panel'
+import { Flex, Text, Button, TextInput, Loader } from '@mantine/core'
+import { isNotEmpty, useForm } from '@mantine/form'
+import { useSignIn } from '@clerk/nextjs'
+import { useMutation } from '@tanstack/react-query'
+import type { SignInResource } from '@clerk/types'
+
+import type { MFAState } from './logic'
+import { errorToString } from '@/lib/errors'
+
+export const RequestMFA: React.FC<{ mfa: MFAState; onReset: () => void }> = ({ mfa, onReset }) => {
+    const { isLoaded, setActive } = useSignIn()
+
+    const form = useForm({
+        initialValues: {
+            code: '',
+        },
+
+        validate: {
+            code: isNotEmpty('Required'),
+        },
+    })
+
+    const { isPending, mutate: onMFASubmit } = useMutation({
+        async mutationFn(form: { code: string }) {
+            if (!isLoaded || !mfa) return
+
+            return mfa.signIn.attemptSecondFactor({
```

Comment:
Not urgent but when we upgrade we'll have access to [useReverification](https://clerk.com/docs/hooks/use-reverification) which might make this a bit nicer down the line